From 91039258fc378c747762806f06b4638430a60188 Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Wed, 31 Aug 2016 09:25:40 +0100
Subject: [PATCH 2/5] topology: remove check for same cpuid cannot exist in 2
 different cores

This breaks on recent kernels because the cpu enumation from /sys may not
be in numerical order.  Remove it for now.

Signed-off-by: Colin Ian King <colin.king@canonical.com>
---
 topology.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/topology.c b/topology.c
index e293a59..1e1e774 100644
--- a/topology.c
+++ b/topology.c
@@ -149,9 +149,6 @@ int add_topo_info(struct cpu_topology *topo_list, struct topology_info *info)
 	if (s_core->cpu_num > 1)
 		s_core->is_ht = true;
 
-	/* Assumption: Same cpuid cannot exist in 2 different cores */
-	assert(!check_exist_from_head(&s_phy->cpu_enum_head, info->cpu_id));
-
 	/* Add to the list (really a set) of all contained cpus in s_phy */
 	list_add_tail(&s_cpu->list_phy_enum, &s_phy->cpu_enum_head);
 
-- 
2.9.3

